src/Geometry/coordinates.jl

```diff
@@ -1,9 +1,40 @@
 abstract type AbstractPoint{FT} end
 
+"""
+    float_type(T)
+
+Return the floating point type backing `T`: `T` can either be an object or a type.
+"""
+float_type(::Type{<:AbstractPoint{FT}}) where {FT} = FT
+float_type(::AbstractPoint) where {FT} = FT
```

Comment:
is this the new domain "undertype"? we should probably make it part of the public API then

Reply:
i'd like to leave that to another PR